includes/class-sensei-learners-main.php

```diff
@@ -263,14 +264,50 @@ protected function get_row_data( $item ) {
 
 				if ( 'complete' == $user_activity->comment_approved || 'graded' == $user_activity->comment_approved || 'passed' == $user_activity->comment_approved ) {
 
-					$status_html = '<span class="graded">' . esc_html__( 'Completed', 'sensei-lms' ) . '</span>';
+					$progress_status_html = '<span class="graded">' . esc_html__( 'Completed', 'sensei-lms' ) . '</span>';
 
 				} else {
 
-					$status_html = '<span class="in-progress">' . esc_html__( 'In Progress', 'sensei-lms' ) . '</span>';
+					$progress_status_html = '<span class="in-progress">' . esc_html__( 'In Progress', 'sensei-lms' ) . '</span>';
 
 				}
 
+				$is_user_enrolled  = Sensei_Course::is_user_enrolled( $this->course_id, $user_activity->user_id );
+				$course_enrolment  = Sensei_Course_Enrolment::get_course_instance( $this->course_id );
+				$enrolment_results = $course_enrolment->get_enrolment_check_results( $user_activity->user_id );
+
+				$enrolment_tooltip_html = '';
+				if ( Sensei()->feature_flags->is_enabled( 'enrolment_provider_tooltip' ) ) {
+					if ( $enrolment_results && ! empty( $enrolment_results->get_provider_results() ) ) {
+						$enrolment_tooltip_html   = [];
+						$enrolment_tooltip_html[] = '<ul class="enrolment-helper">';
+						foreach ( $enrolment_results->get_provider_results() as $id => $result ) {
+							$name = Sensei_Course_Enrolment::get_enrolment_provider_name_by_id( $id );
+							if ( ! $name ) {
+								$name = $id;
+							}
+
+							$item_class = 'does-not-provide-enrolment';
+							if ( $result ) {
+								$item_class = 'provides-enrolment';
+							}
+
+							$enrolment_tooltip_html[] = '<li class="' . esc_attr( $item_class ) . '">' . $name . '</li>';
+						}
+						$enrolment_tooltip_html[] = '</ul>';
+
+						$enrolment_tooltip_html = implode( '', $enrolment_tooltip_html );
+					} else {
+						$enrolment_tooltip_html = esc_html__( 'No enrollment data was found.', 'sensei-lms' );
+					}
+				}
+
+				if ( $is_user_enrolled ) {
+					$enrolment_status_html = '<span class="enrolled sensei-tooltip" data-tooltip="' . esc_attr( htmlentities( $enrolment_tooltip_html ) ) . '">' . esc_html__( 'Enrolled', 'sensei-lms' ) . '</span>';
+				} else {
+					$enrolment_status_html = '<span class="not-enrolled sensei-tooltip" data-tooltip="' . esc_attr( htmlentities( $enrolment_tooltip_html ) ) . '">' . esc_html__( 'Not Enrolled', 'sensei-lms' ) . '</span>';
```

Comment:
Minor nit, but we could avoid this bit of code duplication by putting the class and text in variables and only writing the HTML for both cases once.

Reply:
7e7ae6efc94a5491c89bb25b1b3d56911473aed7